drivers/usb/device/Kconfig

```diff
@@ -40,6 +40,15 @@ config USB_DC_RPI_PICO
 	help
 	  Enable USB support on the RP2 family of processors.
 
+if USB_DC_RPI_PICO
+
+config USB_DC_RPI_PICO_HW_VBUS_DETECT
```

Comment:
There is no need to introduce Kconfig option. It just always needs to be handled.

Reply:
Bus powered devices do not necessarily have to handle this. After being powered the usb initialization will set the pull up
and the device is detected properly. So maybe this should only be handled if `CONFIG_USB_SELF_POWERED` is set?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/78484#discussion_r1768516872

Reply:
But there are some boards that do not even have a vbus detection circuit in hardware (for example the Raspnerry Pi Pico). How should this be handled?

Reply:
Like we usually handle it, see (This is what your PR should look like. Not tested):
https://github.com/jfischer-no/zephyr/commit/302c74ffbeefc6a01ebd97c7c008dd0234cb714c
https://github.com/jfischer-no/zephyr/commit/7fde52b295c9665fb3d204d589c8fa32f55a1bd5